src/constbuffer.c

```diff
@@ -14,16 +11,15 @@
 typedef struct CONSTBUFFER_HANDLE_DATA_TAG
 {
     CONSTBUFFER alias;
-}CONSTBUFFER_HANDLE_DATA;
-
-DEFINE_REFCOUNT_TYPE(CONSTBUFFER_HANDLE_DATA);
+    COUNT_TYPE count;
+} CONSTBUFFER_HANDLE_DATA;
```

Comment:
please add static compile time validation for the existence of count type here, in lieu of DEFINE_REFCOUNT_TYPE. It also helps clarify the requirements of a "REF type" in our code.

Reply:
I am not sure I understand this comment.
Are you suggesting havinf an if defined(COUNT_TYPE)?
If so the only thing we could do in case the type is not defined is to #error.

But the compiler would also error if COUNT_TYPE is not defined.

So the check would add no value...

Or maybe you are referring to another type of check. Can you please clarify?

---
In reply to: [221837290](https://github.com/Azure/azure-c-shared-utility/pull/198#discussion_r221837290) [](ancestors = 221837290)

Reply:
This is a minor suggestion to codify the relationship between the declared type (CONSTBUFFER) and the REF macros. Since we aren''t replacing all instances of REFTYPE_COUNT_CREATE

Reply:
Is the suggestion to add the following code?

#if defined(COUNT_TYPE)
#error COUNT_TYPE not defined
#endif

We do not need to add this as it does not add any value (compiler detects the error by the fact that COUNT_TYPE does not exist).

CONST_BUFFER uses the COUNT_TYPE and the INC_REF_VAR/DEC_REF_VAR/INIT_REF_VAR macros in order to increment/decrement/init the ref count in a platform agnostic way. It does not use the rest of the macros in refcount which are meant for wrapping a structure in a ref counted type..

Or if your suggestion is another, can you please draft it? Thanks!

---
In reply to: [222053358](https://github.com/Azure/azure-c-shared-utility/pull/198#discussion_r222053358) [](ancestors = 222053358)